src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -4397,12 +4404,12 @@ protected RubyBaseNodeWithExecute coercePatternToRope(RubyBaseNodeWithExecute pa
         }
 
         @Specialization(
-                guards = "singleByteOptimizableNode.execute(stringRope)")
+                guards = "singleByteOptimizableNode.execute(stringRope, " +
+                        "getContext().getEncodingManager().getRubyEncoding(stringRope.encoding.getIndex()))")
```

Comment:
This should be replaced by passing the RubyEncoding explicitly if possible (for efficiency & propagating the correct encoding).